test: Fix logger for export import test

danieljbruce opened the pull request:
The change involves the following:
- Adding a change to the delay function to use a variable on the mocha test context instead of the currentRetry function which currently does not correspond to a function that lives on the context
- Adding a setup function which sets up this variable each time any test is run that uses the delay function
- Adding a test for these two functions to document and ensure they behave in such a way that console.info is actually called and the test is retried the appropriate number of times.

Fixes [#1143 ](https://github.com/googleapis/nodejs-datastore/issues/1143)

The console.info logs will further help diagnose flakey tests in https://github.com/googleapis/nodejs-datastore/issues/1122

danieljbruce committed:
A bunch of console logs

Console logs should report how far the program made it.

danieljbruce committed:
Changes to track the current retry properly

The current test fails silently on test.currentRetry(). What we should do instead is store the current attempt on the mocha test.

danieljbruce committed:
Introduce a setup for delay function and test flow

The setup for delay function creates a setup for the mocha test to use the delay function. A test is added to make sure that the delay function works properly.

danieljbruce committed:
Remove unnecessary logging from tests

Some logging is in place that does not need to be there which was useful when debugging the problem. This logging should be removed and the only function should be removed.

danieljbruce committed:
Remove console logs that occur in import/export

The console logs from the import/export test should be removed so that the test doesn’t emit any logs as before.

danieljbruce committed:
Remove a before import log

Remove a log that was leftover from doing testing

danieljbruce committed:
For console.info the input type is string

any should not be used if we can avoid it. string type needs to be used for the argument of the function.

bhshkh commented on system-test/datastore.ts:
```diff
@@ -1323,18 +1323,61 @@ describe('Datastore', () => {
+
+      it('should be sure that the delay function emits console info messages', async function () {
+        // Override console.info to track the number of times it is called.
+        console.info = consoleInfoFunction;
```
Why is this required if the next line is overriding this assignment?

danieljbruce replied:
I think this was a typo which needs to be removed.

danieljbruce committed:
PR fix - remove unnecessary console log

console log does not need to be assigned twice in a row. This is a typo from what was there before.

danieljbruce committed:
Remove only

only should not be used in the system tests

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-datastore into fix-logger-for-export-import-test

# Conflicts:
#	system-test/datastore.ts

danieljbruce committed:
Merge branch 'main' into fix-logger-for-export-import-test

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-datastore into fix-logger-for-export-import-test

danieljbruce committed:
Merge branch 'fix-logger-for-export-import-test' of https://github.com/danieljbruce/nodejs-datastore into fix-logger-for-export-import-test

danieljbruce committed:
Correct the race condition in the test

This test shouldn’t be flakey. It should always pass. We want to correct the race condition in this test.

danieljbruce committed:
Always reset info log count to 0

Reset the log count to account for parameterized tests so that the log count is always set to 0 when it starts a new set of tests.

danieljbruce committed:
Change assertion check to reflect new expectation

Now that the less than or equal to sign has been introduced, the assertion check must be modified accordingly to account for the actual number of times the test runs.

danieljbruce committed:
Use local variable instead of test context variabl

The shared context variable on mocha tests is being shared between tests and causing unexpected results. We should use a local variable instead as it seems to work the same way, but is much simpler to understand.

danieljbruce committed:
linting fix

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

danieljbruce committed:
Merge branch 'main' into fix-logger-for-export-import-test

danieljbruce committed:
Merge branch 'main' into fix-logger-for-export-import-test

danieljbruce committed:
Merge branch 'main' into fix-logger-for-export-import-test

danieljbruce committed:
Merge branch 'main' into fix-logger-for-export-import-test
